library/ctr_drbg.c

```diff
@@ -474,7 +474,8 @@ int mbedtls_ctr_drbg_seed( mbedtls_ctr_drbg_context *ctx,
                   (size_t) ctx->reseed_counter :
                   good_nonce_len( ctx->entropy_len ) );
 
-    ctx->reseed_interval = MBEDTLS_CTR_DRBG_RESEED_INTERVAL;
+    if ( ctx->reseed_interval == 0 )
```

Comment:
Instead of trying to guess with this check on a specific value if mbedtls_ctr_drbg_set_reseed_interval() has been called or not, what about to set the default value of 'reseed_interval` in mbedtls_ctr_drbg_init() and don't do anything here?

Reply:
This makes sense to me. We set the entire ctr_drbg context to 0 in init(), and if we are going to always change the interval from 0 to the default of MBEDTLS_CTR_DRBG_RESEED_INTERVAL in seed() anyways, we might as well do it in init(). I will do the same in hmac_drbg_seed().